### PR TITLE
[Feature] #2040 Show Recipient name if available in Unsigned/History

### PR DESCRIPTION
**Pull request by yenda**
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)
fixing #2040 

### Summary:
PR implements the feature

### Steps to test:
- In transaction history, unsigned transactions and transaction details, transactions with known contacts should show the name of the contact.

[comment]: # (PRs will only be accepted if squashed into single commit.)
status: ready

### CODE REVIEW

File: src/status_im/ui/screens/wallet/transactions/subs.cljs [hunk 1]
```diff
@@ -21,10 +21,25 @@
   (fn [wallet]
     (get wallet :current-tab 0)))
 
+(defn enrich-transaction [{:keys [type to from] :as transaction} contacts]
+  ;; TODO (yenda) proper wallet logic when wallet switching is implemented
+  (let [[contact-address key-contact key-wallet] (if (= type :inbound)
+                                                   [from :from-contact :to-wallet]
+                                                   [to :to-contact :from-wallet])
+        wallet                                   "Main wallet"
```
Comment (rasom) [inline src/status_im/ui/screens/wallet/transactions/subs.cljs]: `"Main wallet"` should be moved to translations?

**Review by rasom — APPROVED** [review]

File: src/status_im/ui/screens/wallet/transactions/subs.cljs (hunk 1 above)
Comment (jeluard) [reply src/status_im/ui/screens/wallet/transactions/subs.cljs]: I understand `Main wallet` is only the default and can be changed by users. This whole part will have to be revamped when moving to multiple wallet anyway.

File: src/status_im/ui/screens/contacts/subs.cljs [hunk 2]
```diff
@@ -4,13 +4,11 @@
             [clojure.string :as str]
             [status-im.bots.constants :as bots-constants]))
 
-(reg-sub
-  :current-contact
+(reg-sub :current-contact
```
Comment (flexsurfer) [inline src/status_im/ui/screens/contacts/subs.cljs:7]: why?

File: src/status_im/ui/screens/wallet/transactions/views.cljs [hunk 3]
```diff
@@ -76,16 +76,16 @@
     (:postponed :pending)   (transaction-icon :icons/arrow-right styles/color-gray4-transparent styles/color-gray7)
     (throw (str "Unknown transaction type: " k))))
 
-(defn render-transaction [{:keys [hash to from type value symbol] :as transaction}]
+(defn render-transaction [{:keys [hash from-contact to-contact to from type value symbol] :as transaction}]
```
Comment (flexsurfer) [inline src/status_im/ui/screens/wallet/transactions/views.cljs:79]: `from-address` ?

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/ui/screens/wallet/transactions/views.cljs (hunk 3 above)
Comment (yenda) [reply src/status_im/ui/screens/wallet/transactions/views.cljs:79]: that's the contact either from-contact or to-contact contains the name of the contact.

File: src/status_im/ui/screens/contacts/subs.cljs (hunk 2 above)
Comment (yenda) [reply src/status_im/ui/screens/contacts/subs.cljs:7]: more consistent derived subscription are always formatted like this, also makes it easier to find-grep

File: src/status_im/ui/screens/wallet/transactions/subs.cljs (hunk 1 above)
Comment (yenda) [reply src/status_im/ui/screens/wallet/transactions/subs.cljs]: good point should I add it or we wait for revamp ?

Comment (jeluard) [reply src/status_im/ui/screens/wallet/transactions/subs.cljs]: Maybe you can introduce a `default-wallet-name` ?

File: src/status_im/ui/screens/wallet/transactions/views.cljs (hunk 3 above)
Comment (flexsurfer) [reply src/status_im/ui/screens/wallet/transactions/views.cljs:79]: i meant to change `to` and `from` on `to-address` and `from-address`

**Review by jeluard — APPROVED** [review]

File: src/status_im/ui/screens/wallet/transactions/subs.cljs (hunk 1 above)
Comment (yenda) [reply src/status_im/ui/screens/wallet/transactions/subs.cljs]: no this should entirely be managed withing wallet-list once it exist, I will make a label for now

Comment (yenda) [reply src/status_im/ui/screens/wallet/transactions/subs.cljs]: the label already exists anyway

File: src/status_im/ui/screens/wallet/transactions/views.cljs (hunk 3 above)
Comment (yenda) [reply src/status_im/ui/screens/wallet/transactions/views.cljs:79]: it is a slightly better terminology but `to` and `from` are the name of the fields in transaction, I think it is better to keep the same names for the same things

File: src/status_im/ui/screens/wallet/transactions/subs.cljs (hunk 1 above)
Comment (jeluard) [reply src/status_im/ui/screens/wallet/transactions/subs.cljs]: Well yes that's why I meanT, this label should be called `default-wallet-name`.

**Comment by rasom** [comment]
@yenda please resolve conflicts

**Pull request merged**